five_state_cohesin/CohesinNetwork.py

```diff
@@ -0,0 +1,180 @@
+
+            rhs_eq[self.state_ids[p]] = sym.collect(eq, self.states[p]).coeff(self.states[p], 1) + rhs_bound
+            rhs_eq[self.state_ids[f"R{p}"]] = sym.collect(eq, self.states[f"R{p}"]).coeff(self.states[f"R{p}"], 1) + rhs_residence
+
+        self.rhs_eq = rhs_eq
+
+
+    def solve_rates(self, parameter_dict_wt):
+
+        rate_dict = {}
+        sol_rates = sym.solve(self.rhs_eq, *self.rates.values())
+
+        for rate, expr in sol_rates.items():
+            value = expr.evalf(subs=parameter_dict_wt)
+            rate_dict[str(rate)] = value
+
+        self.rate_dict = rate_dict
+        self.parameter_dict_wt = parameter_dict_wt
+
+
+    def solve_kinetics(self, parameter_dict, t_max=3600, n_steps=10000):
+
+        if hasattr(self, 'rate_dict'):
+            # Compute model kinetics, starting from fully unbound population of RAD21
+            t = sym.symbols('t')
+            init_state = sym.Matrix([[0]*len(self.bound_states) + list(self.abundances.values())])
+
+            t_span = (0, t_max)
+            t_eval = np.linspace(0, t_max, n_steps)
+
+            y0_eval = list(init_state.evalf(subs=parameter_dict))
+            rhs_eval = list(self.rhs.evalf(subs=self.rate_dict))
+
+            f = sym.lambdify((t, self.states.values()), rhs_eval)
+            solution = scipy.integrate.solve_ivp(f, t_span, y0_eval, t_eval=t_eval)
+
+            return solution
+        
+        else:
+            print("Network lacks fitted rate values - please run 'solve_rates' method first")
+
+
+    def to_extrusion_dict(self, parameter_dict,
+	                      reference_dict={"LEF_transition_rates": {}},
+	                      t=10000,
+	                      site_types=['A'],
+	                      fix_velocity=False,
+	                      active_states=['RN']):
+		
+        output_dict = reference_dict.copy()
+        output_dict["LEF_states"] = {k: v+1 for v, k in enumerate(self.sequence)}
+
+        kinetics = self.solve_kinetics(parameter_dict, t_max=t)
+        final_state = kinetics.y[:,-1]
+
+        for rate, value in self.rate_dict.items():
+            s1 = f"R{rate[-2]}"
+            s2 = f"R{rate[-1]}"
+
+            id1 = self.sequence.index(s1)+1 if s1 in self.sequence else 0
+            id2 = self.sequence.index(s2)+1 if s2 in self.sequence else 0
+
+            rate_key = f"{id1}{id2}"
+		
+            if rate[-1] not in ['R', 'B']:
+                k = self.full_states.index(rate[-1])
+                value *= final_state[k]
+
+            if (id1 != 0) & (id2 != 0):
+                output_dict["LEF_transition_rates"][rate_key] = {t: float(value) for t in site_types}
+            elif id1 == 0:
+                output_dict["LEF_on_rate"] = {t: float(value) for t in site_types}
+            elif id2 == 0:
+                output_dict["LEF_off_rate"] = {t: float(value) for t in site_types}
+                output_dict["LEF_stalled_off_rate"] = {t: float(value) for t in site_types}
+
+        if fix_velocity:
```

Comment:
it would be nice to re-name this bool (as its not broken per se), and it is rather specific to assuming that "N" is the active state. Some possibilities:
- convert_from_two_state
- rescale_velocity (and then the variable inside becomes rescaled_velocity)